packages/modules/devices/sonnen/sonnenbatterie/api.py

```diff
@@ -355,7 +355,7 @@ def update_battery(self, sim_counter: SimCounter) -> BatState:
         # the current is calculated as apparent power / voltage
         battery_ac_voltage = battery_state["Uac"]
         currents = [float(battery_state[f"Sac{phase}"]) / battery_ac_voltage
-                    if battery_state[f"Sac{phase}"] else None
+                    if f"Sac{phase}" in battery_state and battery_state[f"Sac{phase}"] else None
```

Comment:
```suggestion
                    if battery_state.get(f"Sac{phase}) else None
```
Python bietet für Dict `get()`. Wenn es den Key gibt, wird der Wert zurückgegeben, sonst kann man einen default-Rückgabewert definieren, Standard ist None.